core/src/main/scala/pink/cozydev/protosearch/Index.scala

```diff
@@ -71,13 +72,40 @@ sealed abstract class Index private (
 
   /** For every term between left and right, get the docs using those terms. */
   def docsForRange(left: String, right: String): Set[Int] = {
-    import scala.collection.mutable.HashSet
     val bldr = HashSet.empty[Int]
     Range(termIndexWhere(left), termIndexWhere(right))
       .foreach(i => bldr.addAll(evenElems(tfData(i))))
     bldr.toSet
   }
 
+  /** For every term starting with prefix, get the docs using those terms. */
+  def docsForPrefix(prefix: String): Set[Int] = {
+    var i = termIndexWhere(prefix)
+    if (termDict(i).startsWith(prefix)) {
```

Comment:
this is assuming that the prefix value would appear ordered before any full words (cat should appear before cats)

And that should be a safe assumption but also we literally saw `ls` ordering files not like that so lets write a bunch of tests to convince ourselves 😅